pkg/login/access.go

```diff
@@ -0,0 +1,65 @@
+package login
+
+// The kind of login
+// This can be used to access additional metadata fields we might add for a particular login.
+type Kind string
+
+func (x Kind) String() string { return string(x) }
+
+const (
+	KindPassword Kind = "password"
+	KindBearer   Kind = "bearer"
+	KindBasic    Kind = "basic"
+	KindDiscord  Kind = "discord"
+)
+
+// A user role.
+// Users can have multiple roles.
+// These provide different levels of access to the api.
+type Role string
+
+func (x Role) String() string { return string(x) }
+
+const (
+	RoleAnonymous     Role = "anonymous" // anonymous/unauthenticated access to the site
+	RoleVVGOMember    Role = "vvgo-member"
+	RoleVVGOUploader  Role = "vvgo-uploader"
+	RoleVVGODeveloper Role = "vvgo-developer"
+)
+
+var anonymous = Identity{
+	Kind:  "",
+	Roles: []Role{RoleAnonymous},
```

Comment:
Why not bitflags, or a set-like map?

Reply:
I can use bitflags or enum. I just personally find them unreadable/obfuscated.

Reply:
Fair enough.